### DISCUSSION TITLE
Is there a way to remove or insert a bar in the song editor?

### DISCUSSION THREAD

**Discussion by betocool-prog** [hydrogen-music/hydrogen#1916] [Q&A] [answered]
Hey all,

I've been using Hydrogen for the last few years and I'm very happy with it, excellent for good drum kits and jams.

Whenever I'm doing a song, sometimes I realise too late that I missed a bar or that I have a bar too many. Is there a way to remove or insert a blank bar (wedge in) in the middle of a song? I could not find a way, and I end up shifting the bars by clicking the patterns either to the left or the right.

Cheers,

Alberto

**Comment by theGreatWhiteShark** [comment, marked as answer]
Hey @betocool-prog ,

> Is there a way to remove or insert a blank bar (wedge in) in the middle of a song?

Nope.

> I could not find a way, and I end up shifting the bars by clicking the patterns either to the left or the right.

Oh no. You can just drag-select all those patterns with your mouse and then drag-move them to wherever you want all at once. Alternatively, you can also use <kbd>Shift</kbd> and your arrow keys and use <kbd>Ctrl</kbd>+<kbd>x</kbd>/<kbd>v</kbd> to cut and paste them.

**Comment by betocool-prog** [comment]
Thanks, the drag-select functions works well, I didn't know that.

But "shift + arrows" only moves the screen left or right, and CTRL+X, CTRL+V didn't seem to work on a selected group either.

Cheers,

Alberto

**Reply by cme** [reply]
That's... odd. Which version of Hydrogen are you using, and on which platform? That potentially sounds like a very old version?

**Comment by betocool-prog** [comment]
Ah, yes, I was using an older version, sorry about that, I had the version from the package manager which seems pretty old, and the newer appimag (linux) 1.2.2 which works as expected, sorry about the confusion.

Cheers!

Alberto
